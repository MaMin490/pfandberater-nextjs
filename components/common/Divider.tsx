import React from 'react';
import styled from 'styled-components';
import { Divider as DIVIDER } from 'antd';

import { theme } from '../../public/global/colors';
import { device } from '../../public/global/media-query';

const { colors } = theme;

const DIVIDER_CONTAINER = styled.div`
  @media ${device.tablet} {
    margin: 1.5rem 0;
  }

  @media ${device.laptop} {
    margin: 2.5rem 0;
  }
`;

export default function Divider(): JSX.Element {
  return (
    <DIVIDER_CONTAINER>
      <DIVIDER
        orientation="right"
        style={{
          color: colors.mint,
          borderTop: colors.menuGrey,
          position: 'absolute',
          transform: 'translate(-50%, 5.5rem)',
          top: '50%',
          left: '50%',
        }}>
        location
      </DIVIDER>
    </DIVIDER_CONTAINER>
  );
}
