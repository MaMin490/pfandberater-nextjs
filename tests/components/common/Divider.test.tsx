import React from 'react';
import { shallow } from 'enzyme';

import Divider from '../../../components/common/Divider';

describe('Divider', () => {
  it('should render correctly', () => {
    const component = shallow(<Divider />);

    expect(component).toMatchSnapshot();
  });
});
