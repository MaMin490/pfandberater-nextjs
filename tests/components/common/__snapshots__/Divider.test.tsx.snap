// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render correctly 1`] = `
<styled.div>
  <Divider
    orientation="right"
    style={
      Object {
        "borderTop": "rgba(255, 255, 255, 0.65)",
        "color": "#2bebc8",
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translate(-50%, 5.5rem)",
      }
    }
  >
    location
  </Divider>
</styled.div>
`;
